app/scripts/views/settings/account_recovery/confirm_password.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Cocktail = require('cocktail');
+const FormView = require('../../form');
+const PasswordMixin = require('../../mixins/password-mixin');
+const ModalSettingsPanelMixin = require('../../mixins/modal-settings-panel-mixin');
+const Template = require('templates/settings/account_recovery/confirm_password.mustache');
+const PASSWORD_SELECTOR = '#password';
+
+const View = FormView.extend({
+  template: Template,
+  className: 'account-recovery-confirm-password',
+  viewName: 'settings.account-recovery.confirm-password',
+
+  events: {
+    'click .cancel-link': FormView.preventDefaultThen('_cancelPasswordConfirm')
+  },
+
+  _cancelPasswordConfirm() {
+    this.navigate('settings/account_recovery', {
+      hasRecoveryKey: false
+    });
+  },
+
+  setInitialContext(context) {
+    const account = this.getSignedInAccount();
+    const email = account.get('email');
+    context.set({
+      email
+    });
+  },
+
+  submit() {
+    const account = this.getSignedInAccount();
+    const password = this.getElementValue('#password');
+    return account.createRecoveryBundle(password)
+      .then((result) => {
+        this.model.set('recoveryKey', result.recoveryKey);
+        this.showRecoveryKeyView = true;
+        this.navigate('settings/account_recovery/recovery_key', result);
+      })
+      .catch((err) => this.showValidationError(this.$(PASSWORD_SELECTOR), err));
```

Comment:
This will show *any* error as a validation error. That seems wrong to me. An invalid password error should be a validation error, others seem like they should be a general panel error.